main.go

```diff
@@ -384,36 +377,35 @@ func ParseCmdLine() *State {
 		}
 
 		if valid {
-			var proxyUrlFunc func(*http.Request) (*url.URL, error)
-			proxyUrlFunc = http.ProxyFromEnvironment
+			proxyURL := http.ProxyFromEnvironment
 
 			if proxy != "" {
-				proxyUrl, err := url.Parse(proxy)
+				p, err := url.Parse(proxy)
```

Comment:
there were two things with very similar names
so here i use a short name for the most local variable